### PR TITLE
Replace ARGV with Homebrew.args in dev-cmd

### PR DESCRIPTION
**Pull request by mofo37**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Trying a different tack here: replacing `ARGV` in smaller chunks.

### CODE REVIEW

**Comment by scpeters** [comment]
tl;dr: I think `ARGV.named` -> `Homebrew.args.remaining`

I looked through the pipeline log, and I think some of the failures occur on lines that replace `ARGV.named.empty?` with `Homebrew.args.named.empty?`. While glancing at a previous pull request https://github.com/Homebrew/brew/pull/5303 (linked from https://github.com/Homebrew/brew/issues/1860), I noticed that `ARGV.named.empty?` was changed to `args.remaining.empty?` (since it was already in the `module Homebrew` namespace). Maybe that will help?

**Comment by MikeMcQuaid** [comment]
@mofo37 Did you ever manage to run `brew tests` locally? If not, I'd suggest that's the main thing we should help you with that'll unblock your work. Can invite you to our Slack or do a brief screen share or something if that'll help get you unstuck. Thanks again for the PR!

**Comment by mofo37** [comment]
Hey @MikeMcQuaid ! I'd like an invite to slack, thanks! Can you please also invite @n2diving ?

**Comment by MikeMcQuaid** [comment]
@mofo37 I've made a 60 day invite for you to our #first-pr-help room. We need to figure out how to get you to run `brew tests` locally before you do anything else. Please hold off pushing or opening any more PRs until we've done that. Thanks!
